Feature/my questionnaire view

X3n0m0rphose opened the pull request:
A bit tricky with the view, any suggestions to make more space between the questions?

Commit:
add markup for the listview.js

Commit:
modified package.json to enable yarn start outside of docker container

Commit:
Enable new route

	* import and export CreateQuestion view to index.js in Views/Site
	folder
	* add new route for and use CreateQuesiton as component

Commit:
change import order

Commit:
Create initial markup and layout for CreateQuestion view

	* add form and button components fro semantic-ui
	* add func to update state contected to inputs in form
	* add basic layout and styling

Commit:
add new scss file to style.scss and add style.css to .gitignore

Commit:
add divs in ListView.js

Commit:
add func for adding more options

	* create function for adding more options when add options
	  button is clicked

Commit:
Merge branch 'dev' into feature/create-question-view-layout

	merge dev into feature/create-question-view-layout to solve
	problems with ressponsivnes

Commit:
fix layout for all devices

Commit:
remove unused container component

Commit:
reinstate PORT=7771 in start script

Commit:
add missing semicolon

Commit:
add uuid to package.json

Commit:
move func to smaler components

	* remove privius func
	* import question component

	this creates a cleaner view and helps to separete concerns

Commit:
Refactor createquestion func

	* move func for updating option value to CreateOption component
	* rework how new options are added to the form
	* add func for removing options

	This commit improves the overview of the functionality by spliting the code into different
	components and adds func for removing an option

Commit:
improve responsivenes for createQuestion view

Commit:
bugfix question input value unable to update

	* add func for updating state for inputs

	this func was preaviusly removed by mistake, now it's added back in

izabelrosen committed:
Merge pull request #4 from chas-academy/feature/create-question-view-layout

Feature/create question view layout

Commit:
merge branch dev into branch feature/my-questionnaire-view

Commit:
add exact path /my-poll

Commit:
add rout to ListView in my-questionnaire-view in index.js

Commit:
add routes for the ListView.js

Commit:
change filename from my-question.js to  to MyQuestion.js

Commit:
added file MyQuestion.scss

Commit:
change from return<MyQuestion> to return <ListView/> to solve React class method not defined no-undef

Commit:
add <SiteRoute exact path=/my-questionnaire component={Site.MyQuestion} /> in Routes.js

Commit:
changed from {Site.MyQuestion to Site.ViewQuestion in Routes.js

Commit:
fix problem with route

Commit:
add style

Commit:
add style to the MyQuestion.js

Commit:
remove id buttongroup

Commit:
add h2 Created polls

Commit:
remove unused css

Commit:
remove unused css-class

Commit:
remove unused style

SebastianGerS commented on package-lock.json:
```diff
@@ -4,6 +4,18 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
+<<<<<<< HEAD
```
you have some un resolved merge conflict beacous this is a package-lock.json file you might be better of deleting this file and then running npm install

SebastianGerS commented on src/Views/Site/MyQuestion.js:
```diff
@@ -0,0 +1,70 @@
+
```
this is my fault i guess but I was just trying to ressolve the fact that you had not created a view, so we move your component to the view folder, quick and easy, but realy what we should have done is to create a new view and used the just imported this component into that view.

so what you can do here is just move all the code in this document to a new component that you then import into this view (you can look att the code in my CreateQuestion view to get the right structure altought you shulde make it a statless component (which I should correct in mine). just remember to hcange the path to the pole component.

SebastianGerS commented on src/Views/Site/MyQuestion.js:
```diff
@@ -0,0 +1,70 @@
+  render() {
+    const { activeIndex } = this.state
+
+    return (
```
```
return (
     
            <Accordion  fluid className="center-content-grid">
              <div className="margin-tb-2 padding-1">
                <Accordion.Title  className="space-between frame minwidth-10" active={activeIndex === 0} index={0} onClick={this.handleAccordion}>
                    Do you like dogs?
                    <Icon name='dropdown' className="minwidth-10" />
                </Accordion.Title>
                  <Accordion.Content  active={activeIndex === 0}>
                  <Poll />
                  </Accordion.Content>
              </div>
```

.... osv

SebastianGerS commented on src/Components/Poll.js:
```diff
@@ -0,0 +1,37 @@
+  }
+
+  render() {
+    return(
```
```
return(
        <div  className ="center-content-column padding-1">
            <div className="center content-row">
            <button class="ui blue basic button"  >{this.state.active ? "Live results" : "Edit" }</button>
            <button class="ui orange basic button"  onClick={this.togglePoll}>{ this.state.active ? "End" : "Activate" }</button>
            </div>
            { this.state.active &&
            <div class="center-content-column padding-1">
                <h3>Share Poll</h3>
                <div className="center-content-row">
                <button class="ui basic button yellow" >Link</button>
                <button class="ui basic button yellow"  >QR-Code</button>
                </div>
            </div>
            }
          </div>  
        
    );
```

SebastianGerS commented on src/Assets/Styles/Views/Site/MyQuestion.scss:
```diff
@@ -0,0 +1,43 @@
+    align-content: space-between;
+    align-items: center;
+}
+
```
la till följande klasser

```
.center-content-column {
    display:flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}


.frame {
    border: 1px solid #FE9A76;
}

.margin-tb-2 {
    margin: 2em 0;
} 

.minwidth-10 {
    min-width: 10em;
}

.space-between  {
    display: flex;
    flex-direction: row;
    align-items: space-between;
}

.padding-1 {
    padding: 1em;
}
```

SebastianGerS reviewed (commented):
the code I inserted solves some of the problems with the view, but not all of them, but you may use my suggestions as a starting point of fixing the rest of the problems

ghost commented:
Thank you for excellent help!

2018-04-21 0:11 GMT+02:00 SebastianGerS <notifications@github.com>:

> *@SebastianGerS* commented on this pull request.
>
> the code I inserted solves some of the problems with the view, but not all
> of them, but you may use my suggestions as a starting point of fixing the
> rest of the problems
> ------------------------------
>
> In package-lock.json
> <https://github.com/chas-academy/platinum-client/pull/13#discussion_r183166574>
> :
>
> > @@ -4,6 +4,18 @@
>    "lockfileVersion": 1,
>    "requires": true,
>    "dependencies": {
> +<<<<<<< HEAD
>
> you have some un resolved merge conflict beacous this is a
> package-lock.json file you might be better of deleting this file and then
> running npm install
> ------------------------------
>
> In src/Views/Site/MyQuestion.js
> <https://github.com/chas-academy/platinum-client/pull/13#discussion_r183168475>
> :
>
> > @@ -0,0 +1,70 @@
> +
>
> this is my fault i guess but I was just trying to ressolve the fact that
> you had not created a view, so we move your component to the view folder,
> quick and easy, but realy what we should have done is to create a new view
> and used the just imported this component into that view.
>
> so what you can do here is just move all the code in this document to a
> new component that you then import into this view (you can look att the
> code in my CreateQuestion view to get the right structure altought you
> shulde make it a statless component (which I should correct in mine). just
> remember to hcange the path to the pole component.
> ------------------------------
>
> In src/Views/Site/MyQuestion.js
> <https://github.com/chas-academy/platinum-client/pull/13#discussion_r183179506>
> :
>
> > +  }
> +
> +
> +  handleAccordion = (e, titleProps) => {
> +    console.log(e.target);
> +    const { index } = titleProps
> +    const { activeIndex } = this.state
> +    const newIndex = activeIndex === index ? -1 : index
> +
> +    this.setState({ activeIndex: newIndex })
> +  }
> +
> +  render() {
> +    const { activeIndex } = this.state
> +
> +    return (
>
> return (
>
>             <Accordion  fluid className="center-content-grid">
>               <div className="margin-tb-2 padding-1">
>                 <Accordion.Title  className="space-between frame minwidth-10" active={activeIndex === 0} index={0} onClick={this.handleAccordion}>
>                     Do you like dogs?
>                     <Icon name='dropdown' className="minwidth-10" />
>                 </Accordion.Title>
>                   <Accordion.Content  active={activeIndex === 0}>
>                   <Poll />
>                   </Accordion.Content>
>               </div>
>
> .... osv
> ------------------------------
>
> In src/Components/Poll.js
> <https://github.com/chas-academy/platinum-client/pull/13#discussion_r183179708>
> :
>
> > +  constructor(props) {
> +    super(props)
> +
> +    this.state = {
> +      active: false
> +    }
> +
> +    this.togglePoll = this.togglePoll.bind(this)
> +  }
> +
> +  togglePoll() {
> +    this.setState({ active: !this.state.active })
> +  }
> +
> +  render() {
> +    return(
>
> return(
>         <div  className ="center-content-column padding-1">
>             <div className="center content-row">
>             <button class="ui blue basic button"  >{this.state.active ? "Live results" : "Edit" }</button>
>             <button class="ui orange basic button"  onClick={this.togglePoll}>{ this.state.active ? "End" : "Activate" }</button>
>             </div>
>             { this.state.active &&
>             <div class="center-content-column padding-1">
>                 <h3>Share Poll</h3>
>                 <div className="center-content-row">
>                 <button class="ui basic button yellow" >Link</button>
>                 <button class="ui basic button yellow"  >QR-Code</button>
>                 </div>
>             </div>
>             }
>           </div>
>
>     );
>
> ------------------------------
>
> In src/Assets/Styles/Views/Site/MyQuestion.scss
> <https://github.com/chas-academy/platinum-client/pull/13#discussion_r183179781>
> :
>
> > +    align-items: center;
> +    justify-content: center;
> +    flex-wrap: no-wrap;
> +    justify-content: space-between;
> +
> +
> +}
> +.center-content{
> +    display:flex;
> +    flex-direction: row;
> +    flex-wrap: nowrap;
> +    justify-content: center;
> +    align-content: space-between;
> +    align-items: center;
> +}
> +
>
> la till följande klasser
>
> .center-content-column {
>     display:flex;
>     flex-direction: column;
>     align-items: center;
>     justify-content: center;
> }
>
>
> .frame {
>     border: 1px solid #FE9A76;
> }
>
> .margin-tb-2 {
>     margin: 2em 0;
> }
>
> .minwidth-10 {
>     min-width: 10em;
> }
>
> .space-between  {
>     display: flex;
>     flex-direction: row;
>     align-items: space-between;
> }
>
> .padding-1 {
>     padding: 1em;
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chas-academy/platinum-client/pull/13#pullrequestreview-114122424>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AbMc3OJXc-gaf2jkj3u8rVBZ8H0Y9A8Aks5tqlz6gaJpZM4Tcqcw>
> .
>

Commit:
fix layout for listable questionnaires

Commit:
refactor  questionnaires

	* move majority of code from view to component
	* move some code to subcomponent
	* make num of displayed questionnaires dependent of state
	* minor revissions fo func due to splitting into subparts

Commit:
add correct name of commponent to be exported and used for my-questionnaires route

Commit:
change start index to -1

Commit:
remove trailing semi-colon
